File: ios/allyoop/RNUserDefaults.m
//
//  RNUserDefaults.m
//  allyoop
//
//  Created by wayne on 15/12/11.
//  Copyright © 2015年 Facebook. All rights reserved.
//

#import "RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(RNUserDefaults, NSObject)

RCT_EXTERN_METHOD(setObject:(NSString *)key forValue:(NSString *)value callback:(RCTResponseSenderBlock *)cb)

RCT_EXTERN_METHOD(getString:(NSString *)key callback:(RCTResponseSenderBlock *)cb)

RCT_EXTERN_METHOD(removeObject:(NSString *)key callback:(RCTResponseSenderBlock *)cb)

@end

File: ios/allyoop/RNUserDefaults.swift
//
//  RNUserDefaults.swift
//  allyoop
//
//  Created by wayne on 15/12/11.
//  Copyright © 2015年 Facebook. All rights reserved.
//

import Foundation
@objc(RNUserDefaults)

class RNUserDefaults: NSObject {
  
  let defaults: NSUserDefaults = NSUserDefaults.standardUserDefaults()
  
  @objc func setObject(key: String, forValue value: String, callback cb: RCTResponseSenderBlock) -> Void {
    self.defaults.setObject(value, forKey: key)
    cb(["Success"])
  }
  
  @objc func getString(key:String, callback cb: RCTResponseSenderBlock) -> Void {
    if let value = self.defaults.stringForKey(key) {
      cb([value])
    } else {
      cb([false])
    }
  }
  
  @objc func removeObject(key: String, callback cb: RCTResponseSenderBlock) -> Void {
    self.defaults.removeObjectForKey(key)
    cb(["Remove success"])
  }
  
}
